dev/TEAL_opcodes.md

```diff
@@ -1128,3 +1229,74 @@ bitlen interprets arrays as big-endian integers, unlike setbit/getbit
 - Pushes: []byte
 - push a byte-array of length X, containing all zero bytes
 - LogicSigVersion >= 4
+
+## log
+
+- Opcode: 0xb0
+- Pops: *... stack*, []byte
+- Pushes: _None_
+- write bytes to log state of the current application
+- LogicSigVersion >= 5
+- Mode: Application
+
+`log` can be called up to MaxLogCalls times in a program, and log up to a total of 1k bytes.
+
+## tx_begin
+
+- Opcode: 0xb1
+- Pops: _None_
+- Pushes: _None_
+- Begin preparation of a new inner transaction
+- LogicSigVersion >= 5
+- Mode: Application
+
+## tx_field f
```

Comment:
What fields need to be specified?
What fields can be specified?

For example, I guess first valid, last valid, genesis name, genesis ID do not need and must not be specified.
If they are specified, does it panic?

Does it panic at the time an invalid field is inserted (e.g., Amount in an ASA transfer) or only when there is an issue in `tx_submit`?

What happens if you call `tx_field` twice with the same field, changing the field value?

Reply:
Does the sender needs to be specified if the default address is specified?
Can the sender be a rekeyed account to the application address?
I'm assuming it can be in the following comments.

Reply:
The sender is defaulted to the application address.

The sender can be another account if it is rekeyed to the app address.

I need to extend the txn field table to show which fields can be set.  Right now, not much:
Sender
Fee
Receiver
Amount
CloseRemainderTo
Type
TypeEnum
XferAsset
AssetAmount
AssetSender
AssetReceiver
AssetCloseTo

Reply:
As the Fee is defaulted to the least allowable value, taking into account Fee pooling from the group and earlier inner transactions.

Reply:
Details about defaults added.

Reply:
Sorry, I'm not sure where you wrote the details about defaults and authorized fields?